java/libs/src/main/java/org/physical_web/collection/PhysicalWebCollection.java

```diff
@@ -211,4 +216,65 @@ public void jsonDeserialize(JSONObject jsonObject) throws PhysicalWebCollectionE
       addMetadata(pwsResult);
     }
   }
+
+  /**
+   * Return a list of PwPairs sorted by rank in descending order.
+   * @param dedupSiteUrls whether or not to dedup PwPairs by result urls.
+   * @param minRank the minimum rank to keep in the list.
+   * @return a sorted list of PwPairs.
+   */
+  public List<PwPair> getPwPairsSortedByRank(boolean dedupSiteUrls, double minRank) {
+    // Get all valid PwPairs.
+    List<PwPair> allPwPairs = new ArrayList<>();
+    for (UrlDevice urlDevice : mDeviceIdToUrlDeviceMap.values()) {
+      PwsResult pwsResult = mBroadcastUrlToPwsResultMap.get(urlDevice.getUrl());
+      if (pwsResult != null) {
+        allPwPairs.add(new PwPair(urlDevice, pwsResult));
+      }
+    }
+
+    // Sort the list in descending order.
+    Collections.sort(allPwPairs);
+    Collections.reverse(allPwPairs);
+
+    // Filter the list.
+    List<PwPair> ret = new ArrayList<>();
+    Set<String> siteUrls = new HashSet<>();
+    for (PwPair pwPair : allPwPairs) {
+      String siteUrl = pwPair.getPwsResult().getSiteUrl();
+      if (pwPair.getRank() >= minRank
+          && !dedupSiteUrls && !siteUrls.contains(siteUrl)) {
```

Comment:
What if the PwsResults are not equal?  Could have different metadata/rank.  (e.g. in the case that the actual url content changed -- which comes up if you saw one beacon before the second one.  Probably not a very common problem).